generator/src/test/java/com/scottlogic/deg/generator/walker/reductive/ReductiveDecisionTreeReducerTests.java

```diff
@@ -323,6 +323,124 @@ public void shouldReturnNullIfALowLevelDecisionIsInvalidated(){
         Assert.assertThat(reduced, equalTo(null));
     }
 
+
+    @Test
+    public void shouldIncludeAtomicConstraintIfNotContradictoryWithNullForFixedField(){
+        AtomicConstraint field3IsGreaterThan4 = new IsGreaterThanConstantConstraint(field3, 4, Collections.emptySet());
+        TreeConstraintNode field2GreaterThanOrEqualTo2Option = new TreeConstraintNode(
+            Collections.singletonList(isGreaterThanOrEqualTo2),
+            Collections.singletonList(
+                new TreeDecisionNode(
+                    new TreeConstraintNode(field3IsGreaterThan4) //something that isn't yet fixed and itself wouldn't be removed
+                )
+            ));
+        DecisionNode decision1 = new TreeDecisionNode(
+            field2GreaterThanOrEqualTo2Option, //field2 > 2
```

Comment:
As above, update the comment.

```suggestion
            field2GreaterThanOrEqualTo2Option, //field2 >= 2
```

Reply:
thanks, changed.